crates/storage/src/redb_interface.rs

```diff
@@ -0,0 +1,90 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use ream_utils::dir;
+use redb::{Builder, Database, Durability, TableDefinition};
+
+use crate::{config, errors::StoreError};
+
+#[derive(Clone, Debug)]
+pub struct ReamDB {
+    db: Arc<Database>,
+}
+
+impl ReamDB {
+    pub(crate) fn new() -> Result<Self, StoreError> {
+        let ream_dir = dir::create_ream_dir().map_err(StoreError::Io)?;
+
+        let ream_file = ream_dir.join(config::REDB_FILE);
+
+        let db = Builder::new()
+            .set_cache_size(config::REDB_CACHE_SIZE)
+            .create(&ream_file)
+            .map_err(|err| StoreError::Database(err.into()))?;
+
+        Ok(Self { db: Arc::new(db) })
+    }
+
+    fn create_table(&self, name: &str) -> Result<(), StoreError> {
+        let table: TableDefinition<'_, &[u8], &[u8]> = TableDefinition::new(name);
+        let mut txn = self.db.begin_write()?;
+        txn.set_durability(Durability::Immediate);
+        txn.open_table(table)?;
+        txn.commit()?;
+        Ok(())
+    }
+
+    fn acquire_connection(&mut self, table: String) -> Result<Connection, StoreError> {
+        Ok(Connection {
+            backend: Arc::new(self.clone()),
+            table,
+        })
+    }
+}
+
+struct Connection {
```

Comment:
This file is named as interface but actually it's an impl. The original intention would've made sense if we have an interface for multiple backends and impls of each of them.

But I don't think we want to support multiple backend. I'm of the opinion that it won't be our priority for the time being.